react/components/__tests__/QuantitySelector.test.tsx

```diff
@@ -0,0 +1,240 @@
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenLastCalledWith(5)
+    expect(input).not.toBeInTheDocument()
+  })
+
+  it("shouldn't display default measurement unit value", () => {
+    render(
+      <QuantitySelector
+        id="1"
+        value={1}
+        maxValue={50}
+        onChange={jest.fn()}
+        measurementUnit="un"
+      />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toHaveDisplayValue('1')
+  })
+
+  it("shouldn't display raw quantity value when unit multiplier exists", () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector
+        id="1"
+        value={1}
+        maxValue={50}
+        onChange={onChange}
+        unitMultiplier={0.5}
+        measurementUnit="kg"
+      />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toHaveValue('1')
+    expect(dropdown).toHaveDisplayValue('0.5 kg')
+
+    let options = screen.getAllByRole('option')
+
+    // Get the first half options, because we have two
+    // dropdowns in the page
+    options = options.slice(0, options.length / 2)
+
+    expect(options).toHaveLength(12)
+
+    expect(options[1]).toHaveTextContent('0 - Remove')
+    expect(options[1]).toHaveValue('0')
+
+    expect(options[11]).toHaveTextContent('5 kg +')
+    expect(options[11]).toHaveValue('10')
+
+    fireEvent.change(dropdown, { target: { value: '10' } })
+    expect(onChange).toHaveBeenLastCalledWith(10)
+
+    expect(dropdown).not.toBeInTheDocument()
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toHaveValue('5')
+
+    fireEvent.change(input, { target: { value: '6' } })
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenLastCalledWith(12)
+  })
+
+  it('can handle decimal values with both periods and commas with unit multiplier', () => {
+    const onChange = jest.fn()
+    const showToast = jest.fn()
+
+    render(
+      <ToastContext.Provider
+        value={{ showToast, hideToast: jest.fn(), toastState: {} }}
+      >
+        <QuantitySelector
+          id="1"
+          value={10}
+          maxValue={50}
+          onChange={onChange}
+          unitMultiplier={0.5}
+          measurementUnit="kg"
+        />
+      </ToastContext.Provider>
+    )
+
+    const [input] = screen.getAllByRole('textbox')
+
+    fireEvent.change(input, { target: { value: '5.5' } })
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenLastCalledWith(11)
+
+    fireEvent.change(input, { target: { value: '5.6' } })
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenCalledTimes(2)
+    expect(onChange).toHaveBeenLastCalledWith(11)
+
+    expect(showToast).toHaveBeenCalledTimes(1)
+    expect(showToast).toHaveBeenLastCalledWith(
+      'This product is sold by fractions of 0.5kg. Given that, the inserted quantity has been rounded to 5.5kg.'
+    )
+
+    fireEvent.change(input, { target: { value: '6,5' } })
```

Comment:
Could you assert that the toast won't show up in this scenario (that is, a valid fraction but with comma)? I know this isn't happening anymore, but seems fair to add it on the test case.

Reply:
there is a test for this already

https://github.com/vtex-apps/product-list/blob/42a0f6811dabde3a9dbdd49840eb1b9cce76d19e/react/components/__tests__/QuantitySelector.test.tsx#L187-L212

Reply:
oh, didn't realized it was a comma there. my bad

Reply:
should have made it clearer 😆